### PR TITLE
Document conflicting JAI-imageIO issues

### PR DESCRIPTION
**Pull request by hflynn**
See https://trello.com/c/ZEetFk4Z/4-document-conflicting-matlab-jars-re-bug-report-cannot-open-jp2-file-created-by-stereoinvestigator

### CODE REVIEW

**Comment by sbesson** [comment]
@hflynn was hoping to review this but ended up side-tracked to understand the doc link failures in https://github.com/openmicroscopy/bioformats/pull/3029 instead. We will get back to this in 2018.

**Comment by hflynn** [comment]
Listed this for review tomorrow

File: components/autogen/src/format-pages.txt [hunk 1]
```diff
@@ -1123,6 +1123,9 @@ reader = JPEG2000Reader
 writer = JPEG2000Writer
 mif = true
 notes = Bio-Formats uses the `JAI Image I/O Tools <https://github.com/jai-imageio/jai-imageio-core>`_ library to read JP2 files. \n
+Conflicting versions of this no-longer maintained library may cause errors, \n
```
Comment (mtbc) [inline components/autogen/src/format-pages.txt]: looks to me like another hyphen as phrase is "no-longer-maintained"

File: docs/sphinx/developers/components.rst [hunk 2]
```diff
@@ -100,6 +100,11 @@ This is primarily needed for reading images from histology/pathology
 formats in :ref:`formats-gpl <formats-gpl>`.  There are no dependencies
 on other components.
 
+The status of this component means that you may encounter errors due to
+conflicting jars e.g. between Bio-Formats and other toolboxes within Fiji or
```
Comment (mtbc) [inline docs/sphinx/developers/components.rst]: maybe can capitalize these JARs

File: docs/sphinx/users/matlab/index.rst [hunk 3]
```diff
@@ -38,6 +38,15 @@ approximately half the speed of our
 :doc:`showinf command line tool </users/comlinetools/index>`, due to
 overhead from copying arrays.
 
+Troubleshooting
+---------------
+
+If you encounter an error trying to open JPEG-2000 data in MATLAB but the file
+will open e.g. in Fiji using Bio-Formats, it may be due to conflicting
+versions of JAI ImageIO in different jars. As discussed on the component page,
```
Comment (mtbc) [inline docs/sphinx/users/matlab/index.rst]: some more JARs here

File: docs/sphinx/users/matlab/index.rst [hunk 4]
```diff
@@ -38,6 +38,15 @@ approximately half the speed of our
...
+:ref:`JAI ImageIO <forks-jai>` is no longer maintained and you will likely
+need to remove the conflicting jar(s) as a work around.
```
Comment (mtbc) [inline docs/sphinx/users/matlab/index.rst]: workaround (noun)

**Comment by dgault** [comment]
The updated notes for JAI read fine to me. The instance of conflicts that we had seen with FIJI was a very specific case so Im happy having the note in the JPEG-2000, MATLAB and JAI sections. Looks good to merge.

**Pull request merged**